.github/workflows/test.yml

```diff
@@ -46,6 +49,9 @@ jobs:
       # Install Rust, keep in sync with rust-toolchain.toml
       - uses: dtolnay/rust-toolchain@1.87.0
         if: ${{ matrix.component == 'proxy' }}
+      - name: Install poetry
+        run: |
+          PIPX_HOME=/opt PIPX_BIN_DIR=/usr/local/bin/ pipx install poetry==$POETRY_VERSION
```

Comment:
Move this into the same apt-get step at the beginning since it's now online and to match the client job below?